docs/theory/swe.md

```diff
@@ -0,0 +1,342 @@
+cells. Using this notation, we obtain a discrete set of equations for the
+solution in cell $i$ by integrating $(\ref{1})$ over $\Omega_i$ and using
+Green's theorem:
+
+\begin{eqnarray}
+\frac{\partial}{\partial t} \int_{\Omega_i} \mathbf{U} d\Omega_i +
+\int_{\Omega_i} \left[ \frac{\partial\mathbf{E}}{\partial x} +
+\frac{\partial\mathbf{G}}{\partial y} \right] d\Omega_i &=&
+\int_{\Omega_i} \mathbf{S} d\Omega_i \nonumber\\
+\frac{\partial}{\partial t} \int_{\Omega_i} \mathbf{U} d\Omega_i +
+\oint_{\partial\Omega_i} \left( \mathbf{E}~dy - \mathbf{G}~dx \right) &=&
+\int_{\Omega_i} \mathbf{S} d\Omega_i \label{3}
+\end{eqnarray}
+
+This equation can be used to approximate discontinuous flows, because all
+quantities appear under integrals. By contrast, $(\ref{1})$ cannot be used
+where derivatives of $\mathbf{U}$ don't exist.
+
+We can interpret the line integral in ($\ref{3}$) in terms of the flux
+$\mathbf{\vec{F}} = (\mathbf{F}_x, \mathbf{F}_y)$ between a cell $i$ and its
+neighboring cells.
+
+$$
+ \frac{\partial}{\partial t} \int_{\Omega_i} \mathbf{U} d\Omega_i +
+\oint_{\partial\Omega_i} \mathbf{\vec{F}} \cdot \vec{n}~dl =
+\int_{\Omega_i} \mathbf{S} d\Omega_i \label{4}
+$$
+
+Here, we have defined a unit normal vector $\vec{n} = (n_x, n_y)$
+pointing outward along the cell boundary $\partial\Omega_i$. $(\ref{4})$ is a
+"surface integral" with a differential arc length $dl$ integrated over the
+boundary of cell $i$. One obtains this surface integral by integrating $(\ref{2})$
+over the domain $\Omega$ and applying the (two-dimensional) divergence theorem
+to the flux term.
+
+To compare these two integral forms, consider the line integral in $(\ref{3})$
+on a quadrilateral cell along a counterclockwise path connecting its vertices:
+
+\begin{eqnarray}
+I_1 =  &\oint_{\partial\Omega_i}& \left( \mathbf{E}~dy - \mathbf{G}~dx \right) \\
+   = - &\int_{x_1}^{x_2}&\mathbf{G}|_{y_1}~dx ~~~~\mathrm{(bottom)} \\
+     + &\int_{y_1}^{y_2}&\mathbf{E}|_{x_2}~dy ~~~~\mathrm{(right)} \\
+     - &\int_{x_2}^{x_1}&\mathbf{G}|_{y_2}~dx ~~~~\mathrm{(top)} \\
+     + &\int_{y_2}^{y_1}&\mathbf{E}|_{x_1}~dy ~~~~\mathrm{(left)}
+\end{eqnarray}
+
+Compare this to the corresponding term in $(\ref{4})$, a surface integral in
+which the areas (lengths) of the faces (edges) are always positive:
+
+\begin{eqnarray}
+I_2 = &\iint_{\partial\Omega_i}& \mathbf{\vec{F}} \cdot \vec{n}~dl_i \\
+     - &\int_{x_1}^{x_2}&\mathbf{F}_y|_{y_1}~dx ~~~~\mathrm{(bottom)} \\
+     + &\int_{y_1}^{y_2}&\mathbf{F}_x|_{x_2}~dy ~~~~\mathrm{(right)} \\
+     + &\int_{x_1}^{x_2}&\mathbf{F}_y|_{y_2}~dx ~~~~\mathrm{(top)} \\
+     - &\int_{y_1}^{y_2}&\mathbf{F}_x|_{x_1}~dy ~~~~\mathrm{(left)}
+\end{eqnarray}
+
+Evidently $I_1 = I_2$ if $\mathbf{F}_x = \mathbf{E}$ and $\mathbf{F}_y = \mathbf{G}.$
+
+In the rest of this section, we use the flux form $(\ref{4})$ of the shallow
+water equations.
+
+### Finite volume formulation
```

Comment:
How about not having a new subsection, 'Finite volume formulation', because the equations under the 'Spatial Discretization' section have already assumed FV discretization?